src/Exceptions/InvalidReasonPhraseException.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Logiek\Http\Exceptions;
+
+use Exception;
+use Throwable;
+
+class InvalidReasonPhraseException extends Exception implements StatusCodeException
+{
+    public function __construct($reasonPhrase = null, $code = 0, Throwable $previous = null)
+    {
+        $message = 'The HTTP status code is not found by the given reason phrase';
```

Comment:
I feel like we don't have to specify the reason phrase here since the user can find the correspondent reason phrase using the stack trace.